event-buffer/event-buffer-core/src/main/java/uk/gov/justice/services/event/buffer/core/repository/service/ConsecutiveEventBufferService.java

```diff
@@ -0,0 +1,135 @@
+package uk.gov.justice.services.event.buffer.core.repository.service;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.justice.services.event.buffer.api.EventBufferService;
+import uk.gov.justice.services.event.buffer.core.repository.spliterator.EventBufferSpliterator;
+import uk.gov.justice.services.event.buffer.core.repository.streambuffer.StreamBufferEvent;
+import uk.gov.justice.services.event.buffer.core.repository.streambuffer.StreamBufferJdbcRepository;
+import uk.gov.justice.services.event.buffer.core.repository.streamstatus.StreamStatus;
+import uk.gov.justice.services.event.buffer.core.repository.streamstatus.StreamStatusJdbcRepository;
+import uk.gov.justice.services.messaging.JsonEnvelope;
+import uk.gov.justice.services.messaging.JsonObjectEnvelopeConverter;
+import uk.gov.justice.services.messaging.logging.JsonEnvelopeLoggerHelper;
+
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+
+import static java.lang.String.format;
+import static uk.gov.justice.services.messaging.logging.JsonEnvelopeLoggerHelper.toEnvelopeTraceString;
+
+@ApplicationScoped
+@Alternative
+@Priority(2)
+public class ConsecutiveEventBufferService implements EventBufferService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventBufferService.class);
+
+    @Inject
+    private StreamBufferJdbcRepository streamBufferRepository;
+
+    @Inject
+    private StreamStatusJdbcRepository streamStatusRepository;
+
+    @Inject
+    private JsonObjectEnvelopeConverter jsonObjectEnvelopeConverter;
+
+
+    /**
+     *
+     * Takes an incoming event and returns a stream of json envelopes. If the event is not consecutive
+     * according to the stream_status repository then an empty stream is returned and the incomingEvent
+     * is added to the event-buffer-repository. If the incomingEvent is consecutive then it is returned
+     * as a stream with any consecutive events from the buffer. If an event is the first to be processed
+     * for that streamId then the version value must be 1 or the incomingEvent is added to the buffer
+     * and an empty stream is returned.
+     *
+     * @param incomingEvent
+     * @return streamOfConsecutiveEvents
+     */
+    @Override
+    public Stream<JsonEnvelope> currentOrderedEventsWith(final JsonEnvelope incomingEvent) {
+
+        LOGGER.trace(format("Message buffering for message: %s", toEnvelopeTraceString(incomingEvent)));
```

Comment:
toEnvelopeTraceString might be replaceable with the new .toString method, or maybe we can combine the two...